File: frontend/src/types/advisorGroups.ts
import type { WonSalesSummary } from '@/types/dealAnalytics'

export type GroupSource = 'manual' | 'hubspot_team' | 'hubspot_list'

export interface AdvisorGroupMember {
  owner_id: string
  owner_name: string | null
}

export interface AdvisorGroup {
  id: string
  name: string
  description: string | null
  brand_value: string | null
  source: GroupSource
  hubspot_source_id: string | null
  hubspot_source_label: string | null
  members: AdvisorGroupMember[]
  member_count: number
  created_at: string | null
  updated_at: string | null
}

export interface HubSpotTeamOption {
  team_id: string
  team_name: string
  member_count: number
  owner_ids: string[]
}

export interface HubSpotListOption {
  list_id: string
  name: string
  object_type_id: string | null
  processing_type: string | null
  size: number | null
}

import type { ContactAdvisorFields, ContactMethodologyData } from '@/types/contactMetrics'

export interface GroupCompareAdvisorRow extends ContactAdvisorFields {
  owner_id: string | null
  owner_name: string
  open_deals: number
  new_deals_7d: number
  new_deals_30d: number
  stale_45d_open: number
  managed_30d_rate: number | null
  tasks_overdue: number
}

export interface GroupPerformanceMetrics {
  won_sales: WonSalesSummary
  leads_created: WonSalesSummary
  tasks_overdue: number
  tasks_overdue_monthly: WonSalesSummary
  tasks_completed_monthly: WonSalesSummary
  tasks_managed_monthly: WonSalesSummary
  calls_monthly: WonSalesSummary
  whatsapp_monthly: WonSalesSummary
}

export interface GroupCompareRow {
  group_id: string
  group_name: string
  source?: GroupSource
  hubspot_source_label?: string | null
  member_count: number
  assigned_deals: number
  open_deals: number
  new_deals_7d: number
  new_deals_30d: number
  stale_45d_open: number
  tasks_completed: number
  tasks_open: number
  tasks_overdue: number
  deals_with_overdue_tasks: number
  managed_30d_rate: number | null
  contact_methodology?: ContactMethodologyData | null
  performance?: GroupPerformanceMetrics
  advisors: GroupCompareAdvisorRow[]
}

export interface GroupsCompareData {
  brand_value: string
  brand_label: string
  groups: GroupCompareRow[]
}

File: runtime/app/schemas/advisor_groups.py
"""Esquemas para grupos de asesores."""

from typing import Any, Literal

from pydantic import BaseModel, Field

GroupSource = Literal["manual", "hubspot_team", "hubspot_list"]


class AdvisorGroupMemberSchema(BaseModel):
    owner_id: str
    owner_name: str | None = None


class AdvisorGroupSchema(BaseModel):
    id: str
    name: str
    description: str | None = None
    brand_value: str | None = None
    source: GroupSource = "manual"
    hubspot_source_id: str | None = None
    hubspot_source_label: str | None = None
    members: list[AdvisorGroupMemberSchema] = Field(default_factory=list)
    member_count: int = 0
    created_at: str | None = None
    updated_at: str | None = None


class AdvisorGroupCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    description: str | None = None
    brand_value: str | None = None
    source: GroupSource = "manual"
    hubspot_source_id: str | None = None
    hubspot_source_label: str | None = None
    members: list[AdvisorGroupMemberSchema] = Field(default_factory=list)


class AdvisorGroupUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = None
    brand_value: str | None = None
    members: list[AdvisorGroupMemberSchema] | None = None


class HubSpotTeamOptionSchema(BaseModel):
    team_id: str
    team_name: str
    member_count: int
    owner_ids: list[str]


class HubSpotListOptionSchema(BaseModel):
    list_id: str
    name: str
    object_type_id: str | None = None
    processing_type: str | None = None
    size: int | None = None


class GroupsCompareRequest(BaseModel):
    brand_value: str
    group_ids: list[str] = Field(min_length=1, max_length=8)


class GroupCompareRowSchema(BaseModel):
    group_id: str
    group_name: str
    member_count: int
    assigned_deals: int
    open_deals: int
    new_deals_7d: int
    new_deals_30d: int
    stale_45d_open: int
    tasks_completed: int
    tasks_open: int
    tasks_overdue: int
    deals_with_overdue_tasks: int
    managed_30d_rate: float | None = None
    advisors: list[dict[str, Any]] = Field(default_factory=list)
